l10n_br_nfe/models/document.py

```diff
@@ -460,6 +460,19 @@ def atualiza_status_nfe(self, infProt, xml_file):
         )
         self._change_state(state)
 
+    def _exec_after_SITUACAO_EDOC_AUTORIZADA(self, old_state, new_state):
+        self.ensure_one()
+        try:
+            self.make_pdf()
+        except Exception as e:
+            # Não devemos interromper o fluxo
+            # E dar rollback em um documento
+            # autorizado, podendo perder dados.
+            # Se der problema que apareça quando
+            # o usuário clicar no gerar PDF novamente.
+            _logger.error("DANFE Error \n {}".format(e))
```

Comment:
teria que ver, mas esse sistema de hooks atual é meio porco, eu acho que isso vai disparar qdo for uma NFSe tb por examplo. Tb ver se isso nao dispara para as notas de fornecedores, ja tivemos varias tretas assim... teria que verificar isso, talvez filtrar ou alterar o log. Retirei minha aprovação ate isso ser checado melhor.

Reply:
booa raphael, vou melhorar a solução, vou deixar a pr em rascunho por enquanto